sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -368,6 +373,98 @@ public virtual async Task<Response<UploadBlobResult>> UploadBlobAsync(Stream str
             }
         }
 
+        private async Task<ChunkedUploadResult> UploadInChunks(string location, Stream stream, int maxChunkSize, bool async = true, CancellationToken cancellationToken = default)
+        {
+            long length = stream.Length;
+
+            string digest = default;
+            ResponseWithHeaders<ContainerRegistryBlobUploadChunkHeaders> uploadChunkResult = null;
+
+            if (length < maxChunkSize)
+            {
+                digest = OciBlobDescriptor.ComputeDigest(stream);
+
+                uploadChunkResult = async ?
+                    await _blobRestClient.UploadChunkAsync(location, stream, cancellationToken: cancellationToken).ConfigureAwait(false) :
+                    _blobRestClient.UploadChunk(location, stream, cancellationToken: cancellationToken);
+
+                return new ChunkedUploadResult(digest,
+                    uploadChunkResult.Headers.Location,
+                    length);
+            }
```

Comment:
If user passes really large `maxChunkSize` let's say 1GB and the stream length is little less than that something like 999MB, then this condition will be true and we will endup uploading the whole stream at once for a really large stream. 

We might consider setting up some upper limit for `maxChunkSize` to break the stream in chunks.

Reply:
Discussed here: https://github.com/Azure/azure-sdk-for-net/pull/32059#discussion_r1018332277